File: support-portal-frontent/src/app/modules/order/_models/OrderDetailDto.ts
export interface OrderDetailDto {
    OrderId: number;
    SellerName: string;
    OrderNumber: string;
    SaleDate: string;
    CreatedDate: string;
    OrderStatus: string;
    ItemNumber: string;
    ItemTitle: string;
    Sku: string;
    ItemCondition: string | null;
    Supplier: string;
    Quantity: number;
    UnitPrice: number;
    ShippingFee: number;
    SaleTax: number;
    TotalPrice: number;
    ShipToName: string;
    ShipToPhone: string;
    ShipToAddress1: string;
    ShipToAddress2: string;
    ShipToCity: string;
    ShipToState: string;
    ShipToZip: string;
    ShipToCountry: string;
    Purchase: PurchaseDto | null;
    Tracking: TrackingDto | null;
    PurchaseAttempts: PurchaseAttemptDto[] | null;
}

export interface PurchaseDto {
    StartTime: string;
    EndTime: string;
    SupplierAccountId: number;
    SupplierAccount: string;
    SupplierOrderNumber: string;
    EstimatedArrivalTime: string;
    SupplierSubTotal: number;
    SupplierShippingFee: number;
    SupplierDiscount: number;
    SupplierTax: number;
    SupplierTotalPrice: number;
    MarketSaleFee: number;
    MarketAdditionalFeePercentage: number;
    MarketAdditionalFeeFixed: number;
    OrderFee: number;
    ProcessingFee: number;
    Profit: number;
}

export interface TrackingDto {
    OriginalTrackingNumber: string;
    OriginalCarrier: string;
    TrackingNumber: string;
    Carrier: string;
}

export interface PurchaseAttemptDto {
    Id: number;
    StartTime: string;
    EndTime: string | null;
    SupplierAccount: string;
    Status: string;
    Reason: string;
    Note: string;
}

File: support-portal-backend/src/Lumio.SupportPortal.Services/Order/Dtos/OrderDetailDto.cs
namespace Lumio.SupportPortal.Services.Order
{
    public class OrderDetailDto
    {
        public int OrderId { get; set; }
        public string SellerName { get; set; }
        /// <summary>
        /// Unique order number from your market (can be eBay order number, Shopify order number, etc.)
        /// </summary>
        public string OrderNumber { get; set; }
        public string SaleDate { get; set; }
        public DateTime CreatedDate { get; set; }
        public string OrderStatus { get; set; }
        public string ItemNumber { get; set; }
        public string ItemTitle { get; set; }
        public string Sku { get; set; }
        public string? ItemCondition { get; set; }
        public string Supplier { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal ShippingFee { get; set; }
        public decimal SaleTax { get; set; }
        public decimal TotalPrice { get; set; }
        public string ShipToName { get; set; }
        public string ShipToPhone { get; set; }
        public string ShipToAddress1 { get; set; }
        public string ShipToAddress2 { get; set; }
        public string ShipToCity { get; set; }
        public string ShipToState { get; set; }
        public string ShipToZip { get; set; }
        public string ShipToCountry { get; set; }

        public PurchaseDto? Purchase { get; set; }
        public TrackingDto? Tracking { get; set; }
        public List<PurchaseAttemptDto>? PurchaseAttempts { get; set; } = new List<PurchaseAttemptDto>();
    }

    public class PurchaseDto
    {
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int SupplierAccountId { get; set; }
        public string SupplierAccount { get; set; }
        public string SupplierOrderNumber { get; set; }
        public string EstimatedArrivalTime { get; set; }
        public decimal SupplierSubTotal { get; set; }
        public decimal SupplierShippingFee { get; set; }
        public decimal SupplierDiscount { get; set; }
        public decimal SupplierTax { get; set; }
        public decimal SupplierTotalPrice { get; set; }
        public decimal MarketSaleFee { get; set; }
        public decimal MarketAdditionalFeePercentage { get; set; }
        public decimal MarketAdditionalFeeFixed { get; set; }
        public decimal OrderFee { get; set; }
        public decimal ProcessingFee { get; set; }
        public decimal Profit { get; set; }
    }

    public class TrackingDto
    {
        public string OriginalTrackingNumber { get; set; }
        public string OriginalCarrier { get; set; }
        public string TrackingNumber { get; set; }
        public string Carrier { get; set; }
    }

    public class PurchaseAttemptDto
    {
        public int Id { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public int? SupplierAccountId { get; set; }
        public string SupplierAccount { get; set; }
        public string Status { get; set; }
        public string Reason { get; set; }
        public string Note { get; set; }
    }
}
